File: src/handlers.py
import abc


class AbstractHandler(abc.ABC):
    _next_handler: 'AbstractHandler' = None

    def set_next(self, handler: 'AbstractHandler') -> 'AbstractHandler':
        self._next_handler = handler
        return handler

    @abc.abstractmethod
    def filter(self, value: str) -> str:
        raise NotImplementedError

File: src/handlers.js
class AbstractHandler {
  constructor() {
    this._nextHandler = null;
  }

  setNext(handler) {
    this._nextHandler = handler;
    return handler;
  }

  filter(value) {
    throw new Error('Method "filter" must be implemented.');
  }
}

module.exports = AbstractHandler;
